### PR TITLE
Feat/record status updates

### PR DESCRIPTION
**Pull request by juribee**

### CODE REVIEW

File: src/graphql/ApiService.ts [hunk 1]
```diff
@@ -175,14 +168,16 @@ export class ApiService {
...
+  ): Promise<{ id: string }> | void {
+    if (recordIds.length > 0) {
+      const req = this.client.request(UPDATE_RECORD_STATUS, {
+        workbookId: session.meta.workbookId,
+        schemaId: parseInt(session.meta.schemaIds[0], 10),
+        validationState: status,
+        rowIds: recordIds,
+      })
+      return this.handleResponse('queueUpdateRecordStatus', req)
+    }
```
[the remark this answers is not in the document]
Comment (juribee) [reply src/graphql/ApiService.ts:180]: Was thinking about this. In the end, decided to go this way cause in the places where we are using this function we are not doing anything with the returned object.

[the remark this answers is not in the document]
Comment (kimdhendrick) [reply src/graphql/ApiService.ts:180]: I think then we shouldn't return anything in the if. Makes the code safer/less surprising to not have more than one type of return. Either that or a "empty/void" version of the current return type in the else

File: src/lib/RecordChunkIterator.ts [hunk 2]
```diff
@@ -59,15 +59,17 @@ export class RecordChunkIterator extends TypedEventManager<IIteratorEvents> {
...
+    promises.push(this.api.updateRecordStatus(this.session, this.rejectedIds, ERecordStatus.REVIEW))
 
     const acceptedIds = prevChunk.recordIds.filter((recordId) => !rejectedIds.includes(recordId))
-    if (acceptedIds) {
-      await this.api.updateRecordStatus(this.session, acceptedIds, ERecordStatus.ACCEPTED)
-    }
+    promises.push(this.api.updateRecordStatus(this.session, acceptedIds, ERecordStatus.ACCEPTED))
+
+    await Promise.all(promises)
+    this.acceptedIds = this.acceptedIds.concat(acceptedIds)
```
Comment (mairechew) [inline src/lib/RecordChunkIterator.ts:71]: is there a place where we need to access all of the `acceptedIds` / `rejectedIds` at the end? curious the reason for still needing to concat here

Comment (juribee) [reply src/lib/RecordChunkIterator.ts:71]: xD only testing purposes

File: src/graphql/ApiService.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (juribee) [reply src/graphql/ApiService.ts:180]: 🤔 wondering if we still are gonna be able to catch errors if we remove the returns completely from this function...

[the remark this answers is not in the document]
Comment (juribee) [reply src/graphql/ApiService.ts:180]: what u think about ? 
image

[the remark this answers is not in the document]
Comment (kimdhendrick) [reply src/graphql/ApiService.ts:180]: ya, at least it tells the truth now ;-)

**Review by mairechew — APPROVED** [review]

File: src/graphql/ApiService.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kimdhendrick) [reply src/graphql/ApiService.ts:180]: tbh, I'd be happier with no return at all, and the return type is void. And then if/when the need arises to have a return type, that person can figure out what to do in each case.
But not sure everyone would agree with me.

**Review by kimdhendrick — APPROVED** [review]

File: src/graphql/ApiService.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (juribee) [reply src/graphql/ApiService.ts:180]: this last option is not a good idea... we can't await for a function that is not returning a promise.

[the remark this answers is not in the document]
Comment (kimdhendrick) [reply src/graphql/ApiService.ts:180]: I meant `Promise<void>` instead of `Promise<{id: string} | void>`

**Pull request merged**